build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+	if (!installed) {
+		return {
+			result: false,
+			message: "Some dependencies aren't installed. Try installing dependencies manually.",
+			fixCommand: "pnpm install",
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * Returns true if the local git repo has a remote for the microsoft/FluidFramework repo.
+ */
+async function checkHasRemote(context: Context): Promise<CheckResult> {
+	const remote = await context.gitRepo.getRemote(context.originRemotePartialUrl);
+	if (remote === undefined) {
+		return {
+			result: false,
+			message: "No remote found that points to the microsoft/FluidFramework repo.",
+		};
+	}
+	return { result: true };
+}
+
+/**
+ * Returns true if the local git repo has a remote for the microsoft/FluidFramework repo.
+ */
+async function checkBranchUpToDate(context: Context): Promise<CheckResult> {
+	const remote = await context.gitRepo.getRemote(context.originRemotePartialUrl);
+
+	let succeeded = false;
+	try {
+		succeeded = await context.gitRepo.isBranchUpToDate(
+			context.originalBranchName,
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			remote!,
+		);
+	} catch (error) {
+		return {
+			result: false,
+			message: `Error when checking remote branch. Does the remote branch exist? Full error message:\n${
+				(error as Error).message
+			}`,
+		};
+	}
+
+	if (!succeeded) {
+		return {
+			result: false,
+			message: "Branch is out of date with the remote. Try pulling the latest from upstream.",
+			fixCommand: "git pull",
+		};
+	}
+	return { result: true };
+}
+
+/**
+ * Returns true if the release group/package has no prerelease dependencies on other Fluid packages.
+ */
+async function checkHasNoPrereleaseDependencies(
+	context: Context,
+	releaseGroupOrPackage: MonoRepo | Package,
+): Promise<CheckResult> {
+	const { releaseGroups, packages, isEmpty } = await getPreReleaseDependencies(
+		context,
+		releaseGroupOrPackage,
+	);
+
+	const packagesToBump = new Set(packages.keys());
+	for (const releaseGroup of releaseGroups.keys()) {
+		for (const pkg of context.packagesInReleaseGroup(releaseGroup)) {
+			packagesToBump.add(pkg.name);
+		}
+	}
+
+	if (isEmpty) {
+		return {
+			result: true,
+		};
+	}
+
+	return {
+		result: false,
+		message: `Prerelease dependencies found. These release groups and packages must be released and integrated before a release can be done:\n\n${[
+			...releaseGroups.keys(),
+		].join("\n")}\n${[...packages.keys()].join("\n")}`,
+		fixCommand: "pnpm flub bump deps <release group or package>",
+	};
+}
+
+/**
+ * Returns true if there are no policy-check failures.
+ */
+async function checkPolicy(context: Context): Promise<CheckResult> {
+	// policy-check is scoped to the path that it's run in. Since we have multiple folders at the root that represent
+	// the client release group, we can't easily scope it to just the client. Thus, we always run it at the root just
+	// like we do in CI.
+	const result = await execa.command(`npm run policy-check`, {
```

Comment:
Generally I thinks its preferred when invoking applications to specify what you are launching and its arguments array separately as that pattern us much more resistant to OS specific shell parsing polices, shell injection, missing escapes etc.
I don't know of an os that would split off the -check into a flag for example, but if there is on, using more structured data instead of a string would prevent and such issues.

https://github.com/sindresorhus/execa/blob/HEAD/docs/escaping.md provides two nice alternatives with solve this.

```suggestion
	const result = await execa(`npm`, [`run `, `policy-check`], {
```

Reply:
Same applies below as well.